elytron/src/main/java/org/wildfly/extension/elytron/AbstractCredentialStoreResourceDefinition.java

```diff
@@ -184,6 +237,109 @@ protected void removeAliasOperation(OperationContext context, ModelNode operatio
         }
     }
 
+    protected void generateKeyPairOperation(OperationContext context, ModelNode operation, CredentialStore credentialStore) throws OperationFailedException {
+        try {
+            String alias = ALIAS.resolveModelAttribute(context, operation).asString();
+
+            if (credentialStore.exists(alias, KeyPairCredential.class)) {
+                throw ROOT_LOGGER.credentialAlreadyExists(alias, KeyPairCredential.class.getName());
+            }
+
+            String algorithm;
+            ModelNode algorithmModel = ALGORITHM.resolveModelAttribute(context, operation);
+            if (algorithmModel.isDefined()) {
+                algorithm = algorithmModel.asString();
+            } else {
+                algorithm = RSA_ALGORITHM;
```

Comment:
The default value can be defined above using the `SimpleAttributeDefinitionBuilder` instead of doing it here.